src/js/client/editprefset-viewport.js

```diff
@@ -0,0 +1,1001 @@
+            //     timestampCreated: "2018-06-21T19:02:28.057Z",
+            //     timestampUpdated: null
+            // },
+            // {
+            //     id: "alice-usb",
+            //     type: "gpiiKey",
+            //     schemaVersion: "0.1",
+            //     prefsSafeId: "prefsSafe-alice",
+            //     prefsSetId: "gpii-default",
+            //     revoked: false,
+            //     revokedReason: null,
+            //     timestampCreated: "2018-06-21T19:02:28.057Z",
+            //     timestampUpdated: null
+            // }
+        // ],
+
+
+        npsetName: "{that}.options.npset.options.npsetName",  // ""
+        flatPrefs: "{that}.options.prefsSafe.preferences.flat",  // {}
+        commonTerms: "{that}.options.commonTerms",            // []
+        commonTermsSorted: [],
+        contextNames: [],
+        npsetApplications: "@expand:gpii.devpmt.npsetApplications({that}.options.prefsSafe.preferences.flat)", // []
+        allSolutions: "{that}.options.allSolutions", // {}
+
+        unsavedChangesExist: false,
+        unsavedChanges: [],
+
+        // Generic Prefs Filters
+        settingsFilter: "mysettings",
+        settingsSearch: "",
+
+        // A hash which could include an entry for each
+        // settingsTables dynamicComponent that contains
+        // it's local display options (such as mine/all settings),
+        // kept here in case we need to rerender the entire page
+        // for a major prefset change.
+        productTableFilters: {},
+
+        devModeOn: false,
+
+        // Information for if we are currently editing a setting,
+        // and if so, which context/term/product it's for and a
+        // copy of the metadata.
+        currentlyEditing: {
+            active: false,
+            current: {
+                context: "",
+                term: "",
+                value: "",
+                product: ""
+            },
+            metadata: {
+                name: "",
+                description: "",
+                schema: {}
+            }
+        },
+
+        commonTermUsageCounts: {},
+        productTermUsageCounts: [],
+
+        // Contains any free information that pertains to an active dialog.
+        // While we can only display one modal dialog at a time, the options
+        // and data for a dialog can change between instances, such as adding
+        // different products to an NP Set, and presenting a confirm dialog
+        // each time.
+        activeModalDialog: {}
+    },
+    events: {
+        createSettingsTable: null,
+        openBaseDialog: null,
+        openAddContextDialog: null,
+        openEditContextDialog: null,
+        openConfirmDialog: null,
+        openConfirmAddProductDialog: null,
+        openConfirmDeleteContextDialog: null,
+        openConfirmSaveDialog: null,
+        openConfirmRemoveProductDialog: null,
+        startDevWidgets: null
+    },
+    bindings: {
+    },
+    components: {
+        // TESTING
```

Comment:
Shall the comment here and line 172 be removed?

Reply:
these components were actually just for testing, so both base components have been removed.